### ISSUE TITLE
fix(jwt-security): no-decode-without-verify matches any method named decode

### ISSUE THREAD

**Issue by ofri-peretz** [ofri-peretz/eslint#507]
**31 findings.**

## Repro

```js
file.content = file.decode(raw)   // Shopify/cli — a TOML file parser
const accessJwt = sdk.token.decode(accessToken)   // okta — Okta's own method
```

Both report `JWT decoded without verification`. Neither involves `jsonwebtoken`.

The one case that *is* a real `jwt.decode` is `twilio/twilio-node` `TokenAuthStrategy.ts:49` — and Twilio documented why on the line above:

```js
// Decode the token without verifying the signature, as we only
// want to read the expiration for this check
const decoded = jwt.decode(token) as JwtPayload;
```

Reading `exp` to decide when to refresh your own token is standard; the signature is verified server-side.

## Fix direction

Require the callee to resolve to an imported JWT library (`jsonwebtoken`, `jose`, `jwt-decode`) — the SDK-evidence gate this repo already applies elsewhere. A bare method named `decode` is not evidence of anything.

Separately, consider whether the Twilio shape (decode → read `exp` only, no trust decision) warrants a lower severity or an exemption.

## Acceptance

- TDD: the TOML and `sdk.token.decode` cases as `valid`, failing first
- `import jwt from 'jsonwebtoken'; jwt.decode(t)` with no `verify` still reports
